cli/crates/cli/Cargo.toml

```diff
@@ -45,6 +45,7 @@ sysinfo = "0.27"
 tempfile = "3"
 json_dotpath = "1"
 tokio = { version = "1", features = ["full"] }
+regex = "1.7"
```

Comment:
Our scheme is normally:

`dependency = "x"`
`dependency = "0.x"`

Depending on whether the major is over `0`

Reply:
I have a different PR fixing this for other dependencies and sorting them alphabetically

Reply:
I don't mind changing it, but why? If I'm not mistaken `"1.7"` means a version in `[1.7.0, 2.0.0[`. So I don't see why allowing a larger internal `[1.0.0, 2.0.0[` is useful in any way.

Reply:
AFAIK `1.7` means `1.7.0`-`1.7.n`.
In both the CLI and the workers we limit changes to breaking versions, allowing minor versions automatically

Reply:
I think you may be confusing it with "0.x" releases, from the cargo book:
```
1.2.3  :=  >=1.2.3, <2.0.0
1.2    :=  >=1.2.0, <2.0.0
1      :=  >=1.0.0, <2.0.0
0.2.3  :=  >=0.2.3, <0.3.0
0.2    :=  >=0.2.0, <0.3.0
0.0.3  :=  >=0.0.3, <0.0.4
0.0    :=  >=0.0.0, <0.1.0
0      :=  >=0.0.0, <1.0.0
```
https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html

Reply:
I see, I recalled it working differently, perhaps an NPM holdover.
@jakubadamw why do we use this convention?

Reply:
@yoav-lavi mostly because we rarely ever want to denote that we need a version of a dependency higher than a particular minimum minor version. Having that ability would be more useful if we were publishing libraries for other to use (in that case the dependent leaf package could be dependent on the same dependency as one among ours, but with an upper bound, like `<=1.6`, so having a minimum of 1.7 allows us to ensure Cargo won't resolve that dependency to 1.6 if we depend on a feature from 1.7).

However, because all packages are all contained in one workspace (with a single Cargo.lock) and never escape the workspace (i.e. don't get published to crates.io), there's no benefit to denoting lower-bounds on minor or patch versions. If we were following the `1.x` or `0.x.y` scheme, our Cargo.toml would effectively become a mirror of Cargo.lock. The practical downside is that our Renovate PRs would have to update both Cargo.toml *and* Cargo.lock on each update of a dependency (most likely in separate PRs, by the way).

I don't know if this thinking applies to grafbase/grafbase as we do publish the `grafbase` package to crates.io. However, the package is still a leaf executable-only package (not a library to be consumed by others), so I'd say it's a similar situation, actually.